dev-docs/backend.md

```diff
@@ -0,0 +1,54 @@
+# What is backend in inspec?
+
+Inspec::Backend class creates the train connection object. Train lets you talk to your local or remote operating systems and APIs with a unified interface.
+
+It allows you to:
+
+* execute commands via run_command
+* interact with files via file
+* identify the target operating system via os
+* authenticate to API-based services and treat them like a platform
+
+When inspec exec command is run with --target option it establishes the connection to that target using speficifed transport like ssh, winrm etc. By default it uses local transport if the --target is not provided. Backend class is responsible for creating this transprot and establish connection using train.
+
+## Mock backend
+
+Mock is used primarily for testing and debugging. This is used to create a dummy connection.
+InSpec use Mock backend in unit testing environment and some places in code base where backend object is required but it does not require for any real time connection or execution of commands. For example inspec export, check, archive are some of the cli commands uses mock bockend as it required by profile object( https://github.com/inspec/inspec/blob/main/lib/inspec/config.rb#L187) else if not provided it creates the backend with local(https://github.com/inspec/inspec/blob/main/lib/inspec/config.rb#L187) transport which is not required for export, check and archive.
+
+## Where from all Inspec::Backend.create(Inspec::Config.mock) gets called?
+
+  > export
+  > check
+  > archive
+  > in dsl.rb filter_included_controls
+  > profile_vendor.rb in profile_opts
+
+## What Inspec::Backend.create does?
+
+It call unpack_train_credentials of config.rb which returns the hash containing backend, logger and few more details
+example
```

Comment:
Try to explain at the level of intent. "Why" is it doing that. Why is important that it does what it is doing. Highlight the important parts and omit the routine parts. Thanks!